src/gpu/locks.rs

```diff
@@ -0,0 +1,89 @@
+use fs2::FileExt;
+use log::{info, warn};
+use std::fs::File;
+use std::path::PathBuf;
+
+const GPU_LOCK_NAME: &str = "bellman.gpu.lock";
+const PRIORITY_LOCK_NAME: &str = "bellman.priority.lock";
+fn tmp_path(filename: &str) -> PathBuf {
+    let mut p = std::env::temp_dir();
+    p.push(filename);
+    p
+}
+
+#[derive(Debug)]
+pub struct GPULock(File);
+impl GPULock {
+    pub fn lock() -> GPULock {
+        info!("Acquiring GPU lock...");
+        let f = File::create(tmp_path(GPU_LOCK_NAME)).unwrap();
+        f.lock_exclusive().unwrap();
+        info!("GPU lock acquired!");
+        GPULock(f)
+    }
+}
+impl Drop for GPULock {
+    fn drop(&mut self) {
+        info!("GPU lock released!");
+    }
+}
+
+#[derive(Debug)]
+pub struct PriorityLock(File);
+impl PriorityLock {
+    pub fn lock_if_priority(priority: bool) -> Option<PriorityLock> {
+        if priority {
+            info!("Acquiring priority lock...");
+            let f = File::create(tmp_path(PRIORITY_LOCK_NAME)).unwrap();
+            f.lock_exclusive().unwrap();
+            info!("Priority lock acquired!");
+            Some(PriorityLock(f))
+        } else {
+            None
+        }
+    }
+    pub fn is_locked() -> bool {
+        File::create(tmp_path(PRIORITY_LOCK_NAME))
+            .unwrap()
+            .try_lock_exclusive()
```

Comment:
does this acquire the lock if it is called on a non locked version?

Reply:
oh, no, it just checks if that file is locked or not, it doesn't hold the lock